components/brave_rewards/resources/rewards_page/components/connect_account.tsx

```diff
@@ -0,0 +1,275 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import Button from '@brave/leo/react/button'
+import Icon from '@brave/leo/react/icon'
+import ProgressRing from '@brave/leo/react/progressRing'
+import Tooltip from '@brave/leo/react/tooltip'
+
+import { AppModelContext, useAppState } from '../lib/app_model_context'
+import { RouterContext } from '../lib/router'
+import { formatMessage } from '../../shared/lib/locale_context'
+import { useCallbackWrapper } from '../lib/callback_wrapper'
+import { useLocaleContext } from '../lib/locale_strings'
+import { NewTabLink } from '../../shared/components/new_tab_link'
+import { WalletProviderIcon } from '../../shared/components/icons/wallet_provider_icon'
+
+import {
+  ExternalWalletProvider,
+  isSelfCustodyProvider,
+  getExternalWalletProviderName,
+  isExternalWalletProviderAllowed
+} from '../../shared/lib/external_wallet'
+
+import * as routes from '../lib/app_routes'
+import * as urls from '../../shared/lib/rewards_urls'
+
+import { style } from './connect_account.style'
+
+type ConnectState = 'loading' | 'error' | ''
```

Comment:
Does the empty state indicate that the wallet is connected? Or is it ambiguous?

Reply:
This is just the form state - whether to show the spinner or an error message or the normal state. The naming is confusing though; I'll find a better name.

Reply:
Gotcha! Maybe `LoadingState`?